client/components/AllProducts.js

```diff
@@ -1,50 +1,64 @@
 import React from "react";
 import { connect } from "react-redux";
 import { Link } from "react-router-dom";
-import { getProducts } from "..store/products";
+import { getProducts } from "../store/products";
+import { addToCart } from "../store/cart";
+import AddProduct from './AddProduct'
 
 class AllProducts extends React.Component {
   componentDidMount() {
-    this.props.getProducts();
+    this.props.getProducts()
   }
   render() {
+    const {isAdmin} = this.props
     return (
-    <div>
-      <h2>Products</h2>
-    
-     {this.props.products.map (product => (
-     
-     <div key={product.id}>
-     
-      <Link to={`/products/${product.id}`}>
-        <h3>{product.title}</h3>
-          <div>
-          <p>Price: {product.price}</p>
-          <p>In stock: {product.inventoryQty}</p>
+      <div>
+        <h2>Products</h2>
+
+        {isAdmin && <AddProduct />}
+
+        {this.props.products.map((product) => (
+          <div key={product.id}>
+            <Link to={`/products/${product.id}`}>
+              <h3>{product.title}</h3>
+              <div>
+                <p>Price: {product.price}</p>
+                <p>In stock: {product.inventoryQty}</p>
+              </div>
+              <img src={product.photoUrl} />
+            </Link>
           </div>
-        <img src= {product.photoUrl}/>
-      </Link>
-     
+      
+      <p><button onClick={() => this.props.addToCart(product)}>Add {product.title} to Cart</button></p>
+      
      </div>
     
-     ))}
-    </div>
-    );
+        ))}
+      </div>
+    )
   }
 }
 
 const mapStateToProps = (state) => {
   return {
     products: state.products.allProducts,
-  };
-};
+    cart: state.cart,
+    isAdmin: state.auth.isAdmin,
+  }
+}
 
 const mapDispatchToProps = (dispatch) => {
   return {
     getProducts: () => {
-      dispatch(getProducts());
+      dispatch(getProducts())
     },
-  };
-};
+    addToCart: (product) => {
+      dispatch(addToCart(product));
+    },
+    loadInitialData() {
```

Comment:
I don't think you need loadInitialData in here, it's not being used in the component and the same thing is dispatched in the Routes component anyway